Index: gmp-4.3.2+dfsg/mpn/arm/copyd.asm
===================================================================
--- gmp-4.3.2+dfsg.orig/mpn/arm/copyd.asm	2010-02-25 14:23:55.829104622 -0500
+++ gmp-4.3.2+dfsg/mpn/arm/copyd.asm	2010-02-25 14:26:13.079104237 -0500
@@ -54,5 +54,9 @@
 	bne	L(loop)
 	ldmfd	sp!, { r7, r8, r9 }		C restore regs from stack
 L(return):
+.ifdef	USES_THUMB
+	bx	lr
+.else
 	mov	pc, lr
+.endif
 EPILOGUE(mpn_copyd)
Index: gmp-4.3.2+dfsg/mpn/arm/copyi.asm
===================================================================
--- gmp-4.3.2+dfsg.orig/mpn/arm/copyi.asm	2010-02-25 14:23:55.909105745 -0500
+++ gmp-4.3.2+dfsg/mpn/arm/copyi.asm	2010-02-25 14:26:13.079104237 -0500
@@ -49,5 +49,9 @@
 	bne	L(loop)
 	ldmfd	sp!, { r7, r8, r9 }		C restore regs from stack
 L(return):
+.ifdef	USES_THUMB
+	bx	lr
+.else
 	mov	pc, lr
+.endif
 EPILOGUE(mpn_copyi)
Index: gmp-4.3.2+dfsg/mpn/arm/udiv.asm
===================================================================
--- gmp-4.3.2+dfsg.orig/mpn/arm/udiv.asm	2010-02-25 14:23:55.769136246 -0500
+++ gmp-4.3.2+dfsg/mpn/arm/udiv.asm	2010-02-25 14:26:13.079104237 -0500
@@ -50,7 +50,11 @@
 
 	str	n1, [ rem_ptr ]		C store remainder
 	adc	r0, n0, n0		C quotient: add last carry from divstep
+.ifdef	USES_THUMB
+	bx	lr
+.else
 	mov	pc, lr
+.endif
 
 L(_large_divisor):
 	stmfd	sp!, { r8, lr }
